diarymanagement-code-r273/tbaclient/tags/final.rc1/tbaclient/src/TBA/Client/GUI/SettingsDialog.java

```diff
@@ -0,0 +1,751 @@
+        txtNowAlert.setText(nowLocation);
+    }//GEN-LAST:event_btnNowBrowseActionPerformed
+
+    private void fillComboMethods()
+    {
+//        options.clear();
+//        options.add(names[comboName.getSelectedIndex()]);
+
+        for(int x=0; x<5; x++)
+        {
+            boolean found = false;
+            for(int y=0; y<3; y++)
+            {
+                if(methods[x].equalsIgnoreCase(names[y]) == true)
+                {
+                    found = true;
+                }
+            }
+            if (found == false)
+            {
+                options.add(methods[x]);
+            }
+        }
+
+        options.add(methods[5]);
+        ComboBoxModel cbmOptions = new javax.swing.DefaultComboBoxModel(options);
+        comboGesture.setModel(cbmOptions);
+    }
+
+
+    /**
+    * @param args the command line arguments
+    */
+    public static void main(String args[]) {
+        /*java.awt.EventQueue.invokeLater(new Runnable() {
+            public void run() {
+                SettingsDialog dialog = new SettingsDialog(new javax.swing.JFrame(), true);
+                dialog.addWindowListener(new java.awt.event.WindowAdapter() {
+                    public void windowClosing(java.awt.event.WindowEvent e) {
+                        setVisible(false);
+                    }
+                });
+                dialog.setVisible(true);
+            }
+        });*/        
+    }
+
+   // Variables declaration - do not modify//GEN-BEGIN:variables
+   private javax.swing.JPanel ConnectionPanelTab;
+   private javax.swing.JPanel GeneralPanelTab;
+   private javax.swing.JPanel GesturePanel;
+   private javax.swing.JPanel LogonPanel;
+   private javax.swing.JPanel ServerPanel;
+   private javax.swing.JPanel SoundPanel;
+   private javax.swing.JButton btnNowBrowse;
+   private javax.swing.JButton btnOK;
+   private javax.swing.JCheckBox chkNowtAlert;
+   private javax.swing.JComboBox comboGesture;
+   private javax.swing.JComboBox comboName;
+   private javax.swing.JLabel jLabel2;
+   private javax.swing.JLabel jLabel3;
+   private javax.swing.JLabel jLabel4;
+   private javax.swing.JLabel jLabel6;
+   private javax.swing.JLabel jLabel8;
+   private javax.swing.JPanel jPanel12;
+   private javax.swing.JPanel jPanel8;
+   private javax.swing.JTabbedPane jTabbedPane1;
+   private javax.swing.JButton jbCancel;
+   private javax.swing.JTextField txtNowAlert;
+   private javax.swing.JTextField txtPort;
+   private javax.swing.JTextField txtServer;
+   private javax.swing.JTextField txtUser;
+   // End of variables declaration//GEN-END:variables
+
+    /**
+     *  This method reads from settings.ini and populates all the controls with it
+     */
+    public void LoadSettings()
+    {
+
+        File file = new File("settings.ini");
+        if(file.exists())
+        {
+            try {
+                FileReader fstream = new FileReader("settings.ini");
+                BufferedReader in = new BufferedReader(fstream);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *in*. The resource is referenced in statements at the following lines: 521, 525, 526, 528, 536, 540, 542, 545, 548, 554, 556, 560, 564. The resource closure statement is at line: 569. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by BufferedReader.readLine. Either a) close *in* in a try-finally block or b) close the resource by declaring *in* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).